### PR TITLE
doc: release-notes-3.7: Add ADC release notes

### PR DESCRIPTION
**Pull request by decsny**
Add release notes for ADC driver class for 3.7 release.

### CODE REVIEW

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
 
 * ADC
 
+  * Added ADC_DT_SPEC_*BY_NAME() macros to get ADC io-channel information from DT by name.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:272]:
```suggestion
  * Added ``ADC_DT_SPEC_*BY_NAME()`` macros to get ADC IO-channel information from DT by name.
```
maybe better?

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Added support for voltage biasing:
+    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
+      voltage biasing.
+    * Added a `zephyr,vbias-pins` to the adc-controller base binding to describe voltage
+      bias pins.
+    * Implemented for the ADC114s08 ADC driver.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:278]:
```suggestion
    * Implemented TI ADC114s08 ADC driver.
```
better?

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Renamed existing ADC sample to adc_dt.
+  * Added a new sample called adc_sequence that shows more of the runtime adc sequence features.
+  * Added driver for the ENE KB1200.
+  * Added driver for the NXP GAU ADC.
+  * Added I2C bus support for ad559x.
+  * Added support for ad5593.
+  * Added configuration of internal reference voltage value to ad559x to support
+    calls of adc_raw_to_millivolts().
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:286]:
```suggestion
    calls of :c:func:`adc_raw_to_millivolts`.
```
better?

File: doc/releases/release-notes-3.7.rst [hunk 4]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Fixed issue with driver initialization causing improper operation in the ad559x driver
+    regarding the availibility of :kconfig:option:`CONFIG_THREAD_NAME`.
+  * Improved the adc read efficiency and validation in ad559x driver.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:289]:
```suggestion
  * Improved the ADC read efficiency and validation in ad559x driver.
```

File: doc/releases/release-notes-3.7.rst [hunk 5]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Fixed issue with configuration register access in the ads1119 driver.
+  * Enabled acquisition time feature in the NXP LPADC driver.
+  * Supported regulator output as reference to NXP LPADC.
+  * Fixed various issues with dma support in the STM32 ADC driver.
+  * Fixed uninitialized value in kb1200 driver found in static analysis.
+  * Fixed issue with adc_raw_to_millivolts returning half the actual voltage with
+    the tla2021 driver by correcting the reference voltage value.
+  * Supported Nuvoton m2l31x in the numaker ADC driver.
+  * Supported nRF54L15 and nRF54H20 in the nrfx_saadc driver.
+  * Improved the nrf saadc driver by disabling burst mode on unused channels, avoiding freezes.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:301]:
```suggestion
  * Improved the nRF SAADC driver by disabling burst mode on unused channels, avoiding freezes.
```

File: doc/releases/release-notes-3.7.rst [hunk 6]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Added support for power management to the smartbond sdadc and gpadc drivers.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:302]:
```suggestion
  * Added support for power management to the Smartbond SDADC and GPADC drivers.
```

File: doc/releases/release-notes-3.7.rst [hunk 7]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Fixed support for :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` in the smartbond ADC driver.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:303]:
```suggestion
  * Fixed support for :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` in the Smartbond ADC driver.
```

File: doc/releases/release-notes-3.7.rst [hunk 8]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
 
+  * Added ADC_DT_SPEC_*BY_NAME() macros to get ADC io-channel information from DT by name.
+  * Added support for voltage biasing:
+    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
+      voltage biasing.
+    * Added a `zephyr,vbias-pins` to the adc-controller base binding to describe voltage
+      bias pins.
+    * Implemented for the ADC114s08 ADC driver.
+  * Renamed existing ADC sample to adc_dt.
+  * Added a new sample called adc_sequence that shows more of the runtime adc sequence features.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:280]:
```suggestion
  * Added a new sample called adc_sequence that shows more of the runtime ADC sequence features.
```

File: doc/releases/release-notes-3.7.rst [hunk 9]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Added driver for the ENE KB1200.
+  * Added driver for the NXP GAU ADC.
+  * Added I2C bus support for ad559x.
+  * Added support for ad5593.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:284]:
```suggestion
  * Added support for Analog Devices' AD5593.
```

File: doc/releases/release-notes-3.7.rst [hunk 10]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
+      voltage biasing.
+    * Added a `zephyr,vbias-pins` to the adc-controller base binding to describe voltage
+      bias pins.
+    * Implemented for the ADC114s08 ADC driver.
+  * Renamed existing ADC sample to adc_dt.
+  * Added a new sample called adc_sequence that shows more of the runtime adc sequence features.
+  * Added driver for the ENE KB1200.
+  * Added driver for the NXP GAU ADC.
+  * Added I2C bus support for ad559x.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:283]:
```suggestion
  * Added I2C bus support for Analog Devices' AD559x.
```

File: doc/releases/release-notes-3.7.rst [hunk 11]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Added support for ad5593.
+  * Added configuration of internal reference voltage value to ad559x to support
+    calls of adc_raw_to_millivolts().
+  * Fixed issue with driver initialization causing improper operation in the ad559x driver
+    regarding the availibility of :kconfig:option:`CONFIG_THREAD_NAME`.
+  * Improved the adc read efficiency and validation in ad559x driver.
+  * Updated esp32 ADC driver to work with version 5.1 of hal_espressif.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:290]:
```suggestion
  * Updated ESP32 ADC driver to work with version 5.1 of hal_espressif.
```

File: doc/releases/release-notes-3.7.rst [hunk 12]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Added support for DMA mode operation for ESP32S3 and ESP32C3.
+  * Fixed issue with configuration register access in the ads1119 driver.
+  * Enabled acquisition time feature in the NXP LPADC driver.
+  * Supported regulator output as reference to NXP LPADC.
+  * Fixed various issues with dma support in the STM32 ADC driver.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:295]:
```suggestion
  * Fixed various issues with DMA support in the STM32 ADC driver.
```

File: doc/releases/release-notes-3.7.rst [hunk 13]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+  * Fixed uninitialized value in kb1200 driver found in static analysis.
+  * Fixed issue with adc_raw_to_millivolts returning half the actual voltage with
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:297]:
```suggestion
  * Fixed issue with :c:func:`adc_raw_to_millivolts` returning half the actual voltage with
```

File: doc/releases/release-notes-3.7.rst [hunk 14]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
...
+    the tla2021 driver by correcting the reference voltage value.
+  * Supported Nuvoton m2l31x in the numaker ADC driver.
+  * Supported nRF54L15 and nRF54H20 in the nrfx_saadc driver.
+  * Improved the nrf saadc driver by disabling burst mode on unused channels, avoiding freezes.
+  * Added support for power management to the smartbond sdadc and gpadc drivers.
+  * Fixed support for :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` in the smartbond ADC driver.
+
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:304]:
```suggestion
```

**Review by aescolar — DISMISSED** [review]
Thanks @decsny .
Please remember to capitalize acronyms and the first letter of proper nouns :)

File: doc/releases/release-notes-3.7.rst [hunk 15]
```diff
@@ -269,6 +269,41 @@ Drivers and Sensors
 
 * ADC
 
+  * Added ADC_DT_SPEC_*BY_NAME() macros to get ADC io-channel information from DT by name.
+  * Added support for voltage biasing:
+    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
+      voltage biasing.
+    * Added a `zephyr,vbias-pins` to the adc-controller base binding to describe voltage
```
Comment (anangl) [inline doc/releases/release-notes-3.7.rst:276]:
```suggestion
    * Added a ``zephyr,vbias-pins`` property to the adc-controller base binding to describe voltage
```

File: doc/releases/release-notes-3.7.rst (hunk 2 above)
Comment (decsny) [reply doc/releases/release-notes-3.7.rst:278]: I can add TI but this is supposed to be a sub bullet of the voltage biasing addition, I am not sure if I'm doing it right

**Comment by aescolar** [comment]
@decsny please the release notes are not building

**Review by aescolar — DISMISSED** [review]

**Comment by decsny** [comment]
@aescolar @nashif it's fixed

File: doc/releases/release-notes-3.7.rst [hunk 16]
```diff
@@ -390,11 +390,43 @@ Drivers and Sensors
 
 * ADC
 
+  * Added ``ADC_DT_SPEC_*BY_NAME()`` macros to get ADC IO-channel information from DT by name.
+  * Added support for voltage biasing:
+     * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:395]: An empty line above this one is needed

Comment (decsny) [reply doc/releases/release-notes-3.7.rst:395]: @aescolar why?

Comment (decsny) [reply doc/releases/release-notes-3.7.rst:395]: oh, I see there is a weird highlight

**Review by aescolar — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 17]
```diff
@@ -390,11 +390,44 @@ Drivers and Sensors
...
+  * Added support for voltage biasing:
+
+    * Added a :kconfig:option:`CONFIG_ADC_CONFIGURABLE_VBIAS_PIN` selected by drivers that support
+      voltage biasing.
+    * Added a ``zephyr,vbias-pins`` property to the adc-controller base binding to describe voltage
+      bias pins.
+    * Implemented for the TI ADC114s08 ADC driver.
+  * Renamed existing ADC sample to adc_dt.
+  * Added a new sample called adc_sequence that shows more of the runtime
+    :c:struct:adc_sequence features.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:403]:
```suggestion
    :c:struct:`adc_sequence` features.
```

**Comment by aescolar** [comment]
@decsny please rebase

File: doc/releases/release-notes-3.7.rst [hunk 18]
```diff
@@ -390,11 +390,44 @@ Drivers and Sensors
...
+  * Supported regulator output as reference to NXP LPADC.
   * Changed phandle type DT property ``nxp,reference-supply`` to phandle-array type DT property
     ``nxp,references`` in ``nxp,lpc-lpadc`` binding. The NXP LPADC driver now supports passing
     the reference voltage value by using ``nxp,references``.
+  * Fixed various issues with DMA support in the STM32 ADC driver.
+  * Fixed uninitialized value in kb1200 driver found in static analysis.
+  * Fixed issue with :c:func:`adc_raw_to_millivolts` returning half the actual voltage with
+    the tla2021 driver by correcting the reference voltage value.
+  * Supported Nuvoton m2l31x in the numaker ADC driver.
+  * Supported nRF54L15 and nRF54H20 in the nrfx_saadc driver.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:427]:
```suggestion
  * Added support for Nuvoton m2l31x in the numaker ADC driver.
  * Added support for nRF54L15 and nRF54H20 in the nrfx_saadc driver.
```

File: doc/releases/release-notes-3.7.rst [hunk 19]
```diff
@@ -390,11 +390,44 @@ Drivers and Sensors
...
+  * Added configuration of internal reference voltage value to ad559x to support
+    calls of :c:func:`adc_raw_to_millivolts()`.
+  * Fixed issue with driver initialization causing improper operation in the ad559x driver
+    regarding the availibility of :kconfig:option:`CONFIG_THREAD_NAME`.
+  * Improved the ADC read efficiency and validation in ad559x driver.
+  * Updated ESP32 ADC driver to work with version 5.1 of hal_espressif.
+  * Added support for DMA mode operation for ESP32S3 and ESP32C3.
+  * Fixed issue with configuration register access in the ads1119 driver.
+  * Enabled acquisition time feature in the NXP LPADC driver.
+  * Supported regulator output as reference to NXP LPADC.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:418]:
```suggestion
  * Added support for regulator output as reference to NXP LPADC.
```

**Review by kartben — CHANGES_REQUESTED** [review]
in addition to the inline comments, I think sorting the bullets alphabetically would be nice, as it would naturally group additions / changes / etc together making it easier to go through changes at a glance

**Comment by decsny** [comment]
@kartben  i tried to organize the notes

**Review by kartben — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**